src/mongo/client/dbclientcursorshimcursorid.cpp

```diff
@@ -36,7 +36,9 @@ namespace mongo  {
             cursor.ns = ele["ns"].String();
 
             if (!ele["firstBatch"].eoo()) {
-                iter = BSONObjIterator(ele["firstBatch"].Obj());
+                BSONObj firstBatch = ele["firstBatch"].Obj();
+                objects_left = firstBatch.nFields();
```

Comment:
That is correct, we'd have to walk the fields. Better than copying them out via getOwned right? The first batch can be 4mb so you'd count 100k in the worst case (all of the collections are single characters and have no options)

```js
> 4 * 1024 * 1024 / Object.bsonsize(db.test.runCommand({
    "listCollections": "test", cursor: {}}
)["cursor"]["firstBatch"][0])
```

Result: 135300 with the only collection being named "a"